openedx/core/djangoapps/credentials/management/commands/notify_credentials.py

```diff
@@ -164,12 +172,46 @@ def handle(self, *args, **options):
             'auto' if options['auto'] else 'manual',
         )
 
-        course_keys = self.get_course_keys(options['courses'])
-        if not (course_keys or options['start_date'] or options['end_date'] or options['user_ids']):
-            raise CommandError('You must specify a filter (e.g. --courses= or --start-date or --user_ids)')
+        program_course_run_keys = self._get_course_run_keys_for_programs(options["program_uuids"])
+
+        courses = options["courses"]
+        if not courses:
+            courses = []
+        if program_course_run_keys:
+            courses.extend(program_course_run_keys)
+
+        course_keys = self.get_course_keys(courses)
+        if not (
+            course_keys or
+            program_course_run_keys or
+            options['start_date'] or
+            options['end_date'] or
+            options['user_ids']
+        ):
+            raise CommandError(
+                'You must specify a filter (e.g. --courses, --program_uuids, --start-date, or --user_ids)'
+            )
 
         handle_notify_credentials.delay(options, course_keys)
 
+    def _get_course_run_keys_for_programs(self, uuids):
```

Comment:
Follow-up from above: the leading underscore seems to have a different meaning here?

Reply:
https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles

In PEP8 this is defined as 

> _single_leading_underscore: weak "internal use" indicator. E.g. from M import * does not import objects whose names start with an underscore.

It is a code style used to indicate that a method should not be used outside its scope.